modelci/hub/converter/converter.py

```diff
@@ -39,3 +51,85 @@ def convert(model, src_framework: str, dst_framework: str, **kwargs):
 
     else:
         raise NotImplementedError(f"Conversion from {src_framework} to {dst_framework} is not supported yet")
+
+
+def generate_model_family(
+        model: MLModel,
+        max_batch_size: int = -1
+):
+    model_weight_path = model.saved_path
+    if not Path(model.saved_path).exists():
+        (filepath, filename) = os.path.split(model.saved_path)
+        os.makedirs(filepath)
+        with open(model.saved_path, 'wb') as f:
+            f.write(model.weight.__bytes__())
+    net = load(model_weight_path)
+    build_saved_dir_from_engine = partial(
+        generate_path_plain,
+        **model.dict(include={'architecture', 'framework', 'task', 'version'}),
+    )
+    inputs = model.inputs
+    outputs = model.outputs
+    model_input = model.model_input
+
+    generated_dir_list = list()
+
+    torchscript_dir = build_saved_dir_from_engine(engine=Engine.TORCHSCRIPT)
+    tfs_dir = build_saved_dir_from_engine(engine=Engine.TFS)
+    onnx_dir = build_saved_dir_from_engine(engine=Engine.ONNX)
+    trt_dir = build_saved_dir_from_engine(engine=Engine.TRT)
+
+    # TODO: expose custom settings to usrs
+    def torchfamily(torchmodel: torch.nn.Module, mlconvert: bool):
```

Comment:
Maybe it's better to put those family functions outside the `generate_model_family` function to make it simpler and easy to read.